docs/rfcs/XXX-pull-request-review.md

```diff
@@ -18,69 +18,126 @@ Peer review is also a critical part of the path to acceptance for pull requests
+<img width="339" alt="slack_-_github" src="https://user-images.githubusercontent.com/7910250/46391240-ad4e9a00-c690-11e8-904b-e4cfd2c0f667.png">
+Note: Change "Current pull request"
 
-* The review summary bubble is elided after the first sentence or N characters if necessary.
-* Clicking the review summary bubble opens an `IssueishDetailItem` in the workspace center.
-* Clicking a line comment opens or activates an editor on the referenced file and scrolls to center the comment's line, translated according to local changes if appropriate.
-* Line comments within the review are rendered: _with a vertical blue bar_ before the file has been opened and the corresponding decoration is visible; _with a vertical grey bar_ after the file and decoration have been seen; and _with a vertical green bar_ after the comment has been marked "resolved" with the control on its decoration.
-* The review summary bubble and line comment lists are greyed out if a different `IssueishDetailItem` is activated.
+Clicking a pull request in the list opens a `PullRequestDetailItem` in the workspace center.
 
-If a pending review is present, it appears at the top of the "Reviews" section within this tile:
+For PRs that are not listed in the panel, users can use the `github:open-issue-or-pull-request` command:
 
-![pending-review](https://user-images.githubusercontent.com/378023/46275946-9bd69680-c599-11e8-9889-66c35458286a.png)
+<img width="679" alt="xxx-pull-request-review_md_ ___github_github" src="https://user-images.githubusercontent.com/7910250/46391629-c8baa480-c692-11e8-8010-0cceb69394b1.png">
+
+
+### PullRequestDetailItem
+
+Each `PullRequestDetailItem` opened on a pull request displays the full, multi-file diff associated with the pull request. Review comments are shown within the diff. See ["Comment decorations"](#comment-decorations) for description of review comments.
+
+![screen shot 2018-10-03 at 1 50 18 pm](https://user-images.githubusercontent.com/7910250/46391711-1df6b600-c693-11e8-87f3-ad4cdbe8ebd8.png)
+
+TODO: update mock to have "Start review button" and "Add single comment"
+
+Diffs are editable ONLY if the pull request branch is checked out and the local branch history has not diverged from the remote branch history. Otherwise diffs are not editable.
+
+A panel at the bottom of the pane offers various options for sorting and filtering the diff. It also has a "Review Changes" button.
+
+#### Sort Options
+
+<img width="731" alt="slack_-_github" src="https://user-images.githubusercontent.com/7910250/46392358-f6551d00-c695-11e8-8ed4-c7aa95044b06.png">
+
+The default view is sorted by files. This is akin to the "Files changed" tab on dotcom. It displays the diff for all changed files in the PR.
+
+Sorting by reviews is akin to the review summaries that appear on the "Conversation" tab on dotcom. The comments are displayed grouped by review along with some context lines.
```

Comment:
Also: how were you picturing the transitions between states working? Can we preserve your scroll position or focus somehow, so you can not lose your place within a review... ?

Reply:
> I'm also really confused by the "sorting files by type" functionality in the image.

The "sorting files by type" is similar to [search by language](https://github.com/atom/github/search?l=Less&q=commit) on .com. So you could narrow down the changes to certain languages. To make all these options less overwhelming, it could be hidden behind a search input. Like `language:Less`. And it's definitely just a "nice to have", but might be handy for bigger PRs.

Here a new mockup that is a bit more cleaned up. The radios are replaced by a "segmented" control.

![screen shot 2018-10-04 at 11 07 57 am](https://user-images.githubusercontent.com/378023/46449237-dc1a5c80-c7c5-11e8-9454-88185bece8aa.png)

> I'd just rather find a better verb for the UI. Maybe something like "show"?

An alternative might be "Group by"? Group all changes in a different way.

> Also: how were you picturing the transitions between states working? Can we preserve your scroll position or focus somehow, so you can not lose your place within a review... ?

That's a good point. I assume that's not really possible, at least not always. Because the diff you're currently looking at, might not be part of any review. Maybe we should use tabs? Then the scroll position is not expected to be the same. lol And we came full circle. :rofl:

OK, I _think_ this RFC is mostly trying to say:

1. Let's move away from .com's UI. **Why**: If it looks the same/similar, people also expect the same functionality. Which would be a huge effort.
2. Let's make more use of unique features/workflows that an editor can offer, e.g editable diffs. **Why**: To have more reasons to actually use an editor for code review.

If we compare the mockups from both PRs, they are not _that_ different. The second is more about "making the changes the main thing of a PR" vs. the "Conversation".

![screen shot 2018-10-04 at 12 09 45 pm](https://user-images.githubusercontent.com/378023/46450975-163c2c00-c7cf-11e8-967b-3c5bec311ad3.png)